### PR TITLE
Include deleted users/groups/serviceprincipals for Get-AzureRmRoleAssignment calls

### PR DESCRIPTION
**Pull request by darshanhs90**
Include deleted users/groups/serviceprincipals for Get-AzureRmRoleAssignment calls

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ResourceManager/Resources/Commands.Resources.Test/ScenarioTests/RoleAssignmentTests.cs [hunk 1]
```diff
@@ -44,7 +44,14 @@ public void RaClassicAdmins()
             ResourcesController.NewInstance.RunPsTest("Test-RaClassicAdmins");
         }
 
-        [Fact]
+		[Fact]
```
Comment (cormacpayne) [inline src/ResourceManager/Resources/Commands.Resources.Test/ScenarioTests/RoleAssignmentTests.cs]: @darshanhs90 nit: looks like the indentation is off in this added function

File: src/ResourceManager/Resources/Commands.Resources/Models.Authorization/AuthorizationClientExtensions.cs [hunk 2]
```diff
@@ -27,8 +27,9 @@ internal static class AuthorizationClientExtensions
     {
         public const string CustomRole = "CustomRole";
         public const string AuthorizationDeniedException = "Authorization_RequestDenied";
+		public const string DeletedObject = "Object Deleted";
```
Comment (cormacpayne) [inline src/ResourceManager/Resources/Commands.Resources/Models.Authorization/AuthorizationClientExtensions.cs]: @darshanhs90 same comment about the indentation being a little off in the added code

**Comment by cormacpayne** [comment]
On-demand: https://azuresdkci.westus2.cloudapp.azure.com/job/powershell-demand/755/

**Comment by darshanhs90** [comment]
Build failing due to unrelated chagnes like Commands.Aks.Test tests failing

**Pull request merged**